template/YYYY-MM-threat-model-template.md

```diff
@@ -0,0 +1,111 @@
+# Title
+
+- **Author**: Your name (@YourGitHubUserName)
+
+## Overview
+
+<!--
+Provide a succinct high-level description of the system, feature, or component being analyzed. Explain why a threat model is being created for this feature and what the main security goals are. This section should be one to three paragraphs long and understandable by someone outside the development team.
+-->
+
+## System Description
+
+<!--
+Provide a detailed description of the system or feature being modeled. Include information about data flow, key components, and interactions with other systems. Diagrams are encouraged to clarify the architecture and flow of data.
+-->
+
+### Architecture Diagram
+
+<!--
+Include a diagram of the system architecture, showing how different components interact. Highlight any areas where security controls are implemented or where threats might be present.
+-->
+
+## Assets and Security Objectives
+
+<!--
+Identify and describe the key assets that need to be protected (e.g., sensitive data, user credentials, etc.). Define the security objectives for these assets, such as confidentiality, integrity, availability, and privacy.
+-->
+
+### Assets
+
+<!--
+List the assets that are crucial for security, and describe their importance to the system.
+-->
+
+### Security Objectives
+
+<!--
+Define the security objectives for each asset. What are you trying to protect, and why? Include goals such as confidentiality, integrity, availability, and privacy.
+-->
+
+## Threats and Vulnerabilities
+
+<!--
+Identify potential threats and vulnerabilities in the system. Consider various types of threats, such as network attacks, data breaches, insider threats, and more. For each threat, describe how it could impact the system and the likelihood of it occurring.
+-->
+
+### Threat List
+
+<!--
+Provide a list of identified threats. For each threat, describe the potential impact, the likelihood, and any existing vulnerabilities that could be exploited.
+-->
+
+### Vulnerability Analysis
+
+<!--
+Analyze the system for vulnerabilities that could be exploited by the identified threats. Consider common attack vectors, such as SQL injection, cross-site scripting (XSS), and weak authentication mechanisms.
+-->
+
+## Mitigations
+
+<!--
+Describe the security measures and controls that will be implemented to mitigate the identified threats and vulnerabilities. Explain how these mitigations will reduce the risk to an acceptable level.
+-->
```

Comment:
Suggestion: the mitigations should be listed alongside each threat, along with their implemented/planned state.